csrc/scheduler/tools/inlining.cpp

```diff
@@ -193,6 +195,21 @@ size_t MaxPosCalculator::getMaxProducerPosFromConsumer(
     }
     return producer->nDims();
   } else {
+    std::unordered_set<ValGroup> loop_path_groups;
```

Comment:
Since an empty set can mean it's just not set or there's indeed no val group, I think this would be clearer and less error-prone:

```suggestion
    std::optional<std::unordered_set<ValGroup>> loop_path_groups;
```